app/src/main/java/com/example/expensetracker/ExpenseAdapter.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.expensetracker
+
+import android.content.Context
+import android.text.Layout
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.TextView
+import androidx.navigation.findNavController
+import androidx.recyclerview.widget.RecyclerView
+import com.example.expensetracker.database.expense.Expense
+import java.nio.file.Files.size
+
+class ExpenseAdapter(private val context: Context, private val data : exp): RecyclerView.Adapter<ExpenseAdapter.ExpenseviewHolder>() {
+    var expData = arrayOf<exp>()
```

Comment:
You can use [ListAdapter](https://developer.android.com/reference/androidx/recyclerview/widget/ListAdapter), so that you don't have to maintain the list in adapter and it is more performant if the list is getting updated frequently